### PR TITLE
Add localtime strftime

### PR DESCRIPTION
**Pull request by Peter554**
Add a `localtime.strftime` as a convenience for formatting a datetime in the local timezone in a single step (rather than having to call `as_localtime` and then `strftime`).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## v4.4.0 - 2023-10-11
+
+- Add `localtime.strftime` [#100](https://github.com/octoenergy/xocto/pull/100)
+- Add `RangeSet.is_(left|right)_finite` [#96](https://github.com/octoenergy/xocto/pull/96)
```
Comment (Peter554) [inline CHANGELOG.md:6]: Looks like this change didn't get released yet https://github.com/octoenergy/xocto/pull/96

Comment (pydanny) [reply CHANGELOG.md:6]: > Looks like this change didn't get released yet https://github.com/octoenergy/xocto/pull/96

I'll get a release out.

File: xocto/localtime.py [hunk 2]
```diff
@@ -749,3 +749,16 @@ def parse_dt(value: str, tz: zoneinfo.ZoneInfo | None = None) -> datetime_.datet
     """
     _datetime = datetime_.datetime.fromisoformat(value)
     return timezone.make_aware(_datetime, timezone=tz)
+
+
+def strftime(
+    dt: datetime_.datetime,
+    fmt: str,
+    tz: timezone.zoneinfo.ZoneInfo | None = None,
```
Comment (pydanny) [inline xocto/localtime.py:757]:
```suggestion
    tz: zoneinfo.ZoneInfo | None = None,
```

File: setup.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 REPO_ROOT = path.abspath(path.dirname(__file__))
 
-VERSION = "4.3.0"
+VERSION = "4.4.0"
```
Comment (pydanny) [inline setup.py:9]:
```suggestion
VERSION = "4.3.0"
```

A bit easier if we leave version bumps to their own PR. 😄

**Review by pydanny — CHANGES_REQUESTED** [review]
LGTM, just please remove the version bump. 😄

File: xocto/localtime.py (hunk 2 above)
Comment (Peter554) [reply xocto/localtime.py]: 👍  Done

File: setup.py (hunk 3 above)
Comment (Peter554) [reply setup.py]: Okay I removed the version update commit.

**Review by pydanny — APPROVED** [review]
LGTM!
